### ISSUE TITLE
[Bug]: Failing to bundle extension on `shopify app dev`

### ISSUE THREAD

**Issue by asoong** [Shopify/cli#4361]
### Please confirm that you have:

- [x] Searched [existing issues](.) to see if your issue is a duplicate. (If you’ve found a duplicate issue, feel free to add additional information in a comment on it.)
- [x] Reproduced the issue in [the latest CLI version](https://www.npmjs.com/package/@shopify/cli).

### In which of these areas are you experiencing a problem?

Extension

### Expected behavior

It should properly build the extension and load it.

### Actual behavior

Getting the following output below in console when running shopify app dev.

I started off doing this and getting the error in a private repo and got the same error when creating a new app and extension from scratch, detailed in the reproduction steps.

It seems to have trouble building the extension (no dist ouput) despite using the shopify cli to scaffold the extension. Is there a system dependency that the cli depends on to build the extension?

```
[ERROR] Could not resolve "@shopify/ui-extensions-react/admin"

    extensions/test-app/src/ActionExtension.jsx:16:7:
      16 │ } from '@shopify/ui-extensions-react/admin';
         ╵        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

  The Yarn Plug'n'Play manifest forbids importing "@shopify/ui-extensions-react" here because it's
  not listed as a dependency of this package:

    ../../.pnp.cjs:43:33:
      43 │           "packageDependencies": [\
         ╵                                  ~~

  You can mark the path "@shopify/ui-extensions-react/admin" as external to exclude it from the
  bundle, which will remove this error and leave the unresolved path in the bundle.

✘ [ERROR] Could not resolve "react/jsx-runtime"

    extensions/test-app/src/ActionExtension.jsx:23:44:
      23 │ export default reactExtension(TARGET, () => <App />);
         ╵                                             ^

  The Yarn Plug'n'Play manifest forbids importing "react" here because it's not listed as a
  dependency of this package:

    ../../.pnp.cjs:43:33:
      43 │           "packageDependencies": [\
         ╵                                  ~~

  You can mark the path "react/jsx-runtime" as external to exclude it from the bundle, which will
  remove this error and leave the unresolved path in the bundle.

✘ [ERROR] Could not resolve "@shopify/ui-extensions-react/admin"

    extensions/test-app/src/ActionExtension.jsx:16:7:
      16 │ } from '@shopify/ui-extensions-react/admin';
         ╵        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

  The Yarn Plug'n'Play manifest forbids importing "@shopify/ui-extensions-react" here because it's
  not listed as a dependency of this package:

    ../../.pnp.cjs:43:33:
      43 │           "packageDependencies": [\
         ╵                                  ~~

  You can mark the path "@shopify/ui-extensions-react/admin" as external to exclude it from the
  bundle, which will remove this error and leave the unresolved path in the bundle.

✘ [ERROR] Could not resolve "react/jsx-runtime"

    extensions/test-app/src/ActionExtension.jsx:23:44:
      23 │ export default reactExtension(TARGET, () => <App />);
```

### Verbose output

The output is similar, the prior output suggests it successfully sets up a tunnel and then attempts to build the extension.

### Reproduction steps

1. Run `shopify app init`, choose `Remix`, `TypeScript`
2. Run `shopify app generate extension`, choose `Admin action`
3. Run `shopify app dev`

### Operating System

macOS Sonoma 14.5

### Shopify CLI version (check your project's `package.json` if you're not sure)

3.66.0

### Node version (run `node -v` if you're not sure)

v22.5.1

**Comment by gonzaloriestra** [comment]
@md-brethren that looks like a different error. Could you please [open a new issue](https://github.com/Shopify/cli/issues/new?template=bug-report.yml) and add as many details as you can?

**Comment by asoong** [comment]
> Hi! Are you using Yarn 2.x or above? Because the CLI is only compatible with Yarn classic (1.x). Or NPM or PNPM.
> 
> We'll update the requirements [here](https://shopify.dev/docs/api/shopify-cli#requirements), because that was removed by mistake recently.

I'm running yarn 1.22.1

**Comment by isaacroldan** [comment]
Closing for now, if you still have issues post a comment and I'll reopen the issue @asoong

**Comment by aaronkantrowitz** [comment]
Also on yarn @1.22.19 

```
The Yarn Plug'n'Play manifest forbids importing
 "@shopify/ui-extensions-react" here because it's not listed as a dependency of this 
```

**Comment by aaronkantrowitz** [comment]
I found the root cause of the issue. There was a .pnp.cjs file on the root folder. Removing that resolved this issue. It conflicted with Yarns workspace approach. @isaacroldan @asoong

**Comment by prayge** [comment]
If anyone comes looking for this bug with:
Failed to bundle extension YOUR_EXTENSION. Please check the extension source code for errors.

It could be caused by old shopify cli version and/or cache retainment:

npm install -g @shopify/cli@latest
shopify cache clear

**Comment by sshaw** [comment]
> It could be caused by old shopify cli version and/or cache retainment:
> npm install -g @shopify/cli@latest
> shopify cache clear

My app has multiple extensions. I was trying to deploy new features for Extension X and failed to deploy due to Extension Y, which had  not been touched for a loooooong time!

`shopify cache clear` did not fix it. 

I upgraded (sigh) and then I got this error: 

    Invalid filename in your theme app extension: snippets/discount-in-progress.liquid~          
    Only .liquid files are allowed in snippets


This would have been fixed ~it~ by #6018...
